### PR TITLE
Jetty 10.0.x #3290 websocket close

### PR DESCRIPTION
**Pull request by gregw**
This PR includes #3335 from @lachlan-roberts but includes 11dfe70fc25ae3d91ed04751c7fe6723ac3c77ae, which generalises `onClose(Throwable cause)` to all `EndPoint`s and `Connection`s (rather than just have it on the Abstract variants).

Perhaps the onClose should be considered separately to the websocket close handling?  @sbordet thoughts?

### CODE REVIEW

**Comment by lachlan-roberts** [comment]
@sbordet @joakime can we get a review on this, if there are any further changes that need to be made I can pull gregs changes into my PR at #3335 and make the changes

File: jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java [hunk 1]
```diff
@@ -235,6 +263,10 @@ public void onCompleteFailure(Throwable failure)
             entry.release();
         }
         entries.clear();
+        if (endPoint instanceof AbstractEndPoint)
+            ((AbstractEndPoint)endPoint).close(failure);
+        else
+            endPoint.close();
```
Comment (sbordet) [inline jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java]: Now that you have `EndPoint.close(Throwable)`, this `instanceof` is not needed anymore.

Comment (gregw) [reply jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java]: @lachlan-roberts Do you want me to make this change in this branch to save making a new branch and PR?

**Comment by gregw** [comment]
@sbordet please re-review

**Pull request merged**